docs/developing/app-review-guide.md

```diff
@@ -0,0 +1,96 @@
+that were present in the dev environment and wasn't adequately tested prior to
+submission.
+
+The latter concern often occurs when a pure client app is submitted up that
+implements localStorage as its storage medium, which Sandstorm does not effectively
+support. This can often be tested by storing some information in a grain, allowing
+Sandstorm to shut down that grain, and then opening it again and seeing if things
+persisted that an ordinary user would expect to persist between sessions.
+
+Finally, the App Market may reject apps which present security or privacy issues
+that fall outside of what a Sandstorm user would expect when using the app. This
+can mean it requests overly-broad permissions when making Powerbox requests or
+uses client-side loading to load scripts which share unexpected information outside
+of the Sandstorm sandbox. For example, we may block approval on an app which attempts
+to load an analytics script from an outside server, or does not make it suitably
+clear to a user why it needs certain outside network access prior to requesting it.
+
+**Note:** Client-side loading is a known gap in Sandstorm's security model that we
+intend to close. Not only is misuse of it a security or privacy issue, its use may
+lead to apps that break when Sandstorm closes this gap.
+
+## Commercial or proprietary apps
+
+The Sandstorm App Market permits both commercial and proprietary apps, however
+official support for app purchase and licensing is not currently available. For an
+app that is merely closed source, but free, this may not pose an issue. For apps
+which require some form of licensing to function, we expect at minimum that the
+app description clearly and prominently specifies the licensing requirements and
+activation method. We would also encourage the app to have at least a minimal level of
+functionality such as a trial or limited version, such that users can experience the
+app before deciding on a purchase. As this is a territory the Sandstorm app review
+team hasn't spent a lot of time on, interested parties may wish to reach out to the
+Sandstorm community for feedback prior to committing significant work here.
+
+## Copyright and copyleft
+
+The developer is responsible for ensuring their packages comply with all relevant
+copyright or copyleft limitations. If a package is reported to us as in violation of a
+license, it may be pulled from the App Market.
+
+Sandstorm package files (or SPKs) are an archive format which includes a variety of
+components needed to run a given application. Often the process of building an SPK pulls
+in components and libraries from open source distributions such as Debian. We believe
+these fall comfortably within the GPLv3's [system libraries exception](https://www.gnu.org/licenses/quick-guide-gplv3.html#less-source-to-distribute-new-system-libraries-exception) and falls well within the spirit of most open source licensing models.
```

Comment:
I don't think the system libraries exception applies here. The system libraries exception says you're allowed to link against proprietary libraries that any user of the platform is expected to have anyway. That's not what's happening here, though... the spk is bundling its own copies of libraries that the user may or may not have -- after all, the user might not be running Debian.

However, I don't think the system libraries exception is important here. In real life, if you distribute an spk containing Debian libraries, you satisfy the source code redistribution requirements by pointing at Debian's source repositories. Nobody really expects that the developer of a Debian-based container should be required to mirror the Debian source repo.

I don't remember to what extent the GPL is explicit about that vs. it just being unwritten rules of reasonableness...

Reply:
My read on "For example, it now also includes the standard libraries of common programming languages such as Python and Ruby." (which I would argue most software users don't, by default, have in their OS) means that if it's trivial to find and acquire the source of commonly-known packages, we need not include the source for them.

I think I was most interested in how this would apply to proprietary packages, where perhaps the build scripts aren't posted anywhere (as was the case with draw.io before it went open source). But presumably I think the requirements may be satisfied if there are no modifications to the open source binaries, and those are commonly well-known components you can find the source for online. Though as discussed above, I think we probably need to better support unpacking and repacking, such that you could swap out the open source components of a proprietary SPK and repack it (with a different app key, of course).

Reply:
I think the point about Python and Ruby is that the the user of a Python program likely has to install Python first. A person distributing a Python script, then, is not required to provide the complete source code for Python. Python itself is effectively "the OS" here.